pkg/operator/ceph/cluster/mgr/dashboard.go

```diff
@@ -110,52 +105,65 @@ func (c *Cluster) configureDashboardModules() error {
 	return nil
 }
 
-func (c *Cluster) configureDashboardModuleSettings(daemonID string) (bool, error) {
+func (c *Cluster) configureDashboardModuleSettings() (bool, error) {
+
 	monStore := config.GetMonStore(c.context, c.clusterInfo)
 
-	daemonID = fmt.Sprintf("mgr.%s", daemonID)
+	// Remove any existing per mgr-daemon configuration
+	if removeMgrDaemonConfiguration {
```

Comment:
How about if we remove these deletions down below so it is after the new settings are configured? Otherwise there may be a brief time where the old settings are deleted and the new settings are not yet applied.